packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.types.ts

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import { IStyle, ITheme } from '../../Styling';
+import { IStyleFunctionOrObject } from '../../Utilities';
+
+export interface IShimmer {}
+
+/**
+ * Shimmer component props.
+ */
+export interface IShimmerProps extends React.AllHTMLAttributes<HTMLElement> {
+  /**
+   * Optional callback to access the IShimmer interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IShimmer | null) => void;
+
+  /**
+   * Sets the width of the shimmer wave wrapper in percentages relative to the containig parent element.
+   * @default 100%
+   */
+  widthInPercentage?: number;
+
+  /**
+   * Sets the width of the shimmer wave wrapper to an exact value in pixels.
+   * If none of the widths provided, it defaults to 100%.
+   */
+  widthInPixel?: number;
+
+  /**
+   * Controls when the shimmer is swapped with actual data through an animated transition.
+   * @default false
+   */
+  isDataLoaded?: boolean;
+
+  /**
+   * Elements to render in one line of the Shimmer.
+   */
+  shimmerElements?: IShimmerElement[];
+
+  /**
+   * Custom elements when necessary to build complex placeholder skeletons.
+   */
+  customElementsGroup?: React.ReactNode;
+
+  /**
+   * Localized string of the status label for screen reader
+   */
+  ariaLabel?: string;
+
+  /**
+   * Call to provide customized styling that will layer on top of the variant rules.
+   */
+  styles?: IStyleFunctionOrObject<IShimmerStyleProps, IShimmerStyles>;
+
+  /**
+   * Additional CSS class(es) to apply to the Shimmer container.
+   */
+  className?: string;
+
+  /**
+   * Theme provided by High-Order Component.
+   */
+  theme?: ITheme;
+}
+
+/**
+ * Shimmer Elements Interface
+ */
+export interface IShimmerElement {
+  /**
+   * Required for every element you intend to use.
+   */
+  type: ShimmerElementType;
+
+  /**
+   * The height of the element (ICircle, ILine, IGap) in pixels.
+   * Read more details for each specific element.
+   */
+  height?: number;
+
+  /**
+   * The width of the element (ILine, IGap) in pixels.
+   * Read more details for each specific element.
+   */
+  widthInPixel?: number;
+
+  /**
+   * The width of the element (ILine, IGap) in pixels.
+   * Read more details for each specific element.
+   */
+  widthInPercentage?: number;
+
+  /**
+   * The vertical alignemt of the element (ICircle, ILine).
+   * @default center
+   */
+  verticalAlign?: ShimmerElementVerticalAlign;
```

Comment:
I'm curious if it would be better to use strings than enum values

Reply:
If you create a codepen using raw javascript you'll see what i mean

Reply:
Fixed.